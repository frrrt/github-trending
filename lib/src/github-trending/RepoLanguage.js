import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@material-ui/core';
import LanguageColors from '../../languageColors';

const useStyles = makeStyles(theme => ({
    root: {
        display: 'flex',
        alignItems: 'center',
    },
    circle: {
        borderRadius: theme.spacing(1),
        width: theme.spacing(2),
        height: theme.spacing(2),
        marginRight: theme.spacing(1),
    },
}));

const RepoLanguage = ({ language }) => {
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <div className={classes.circle} style={{ backgroundColor: LanguageColors[language] ?? (language && '#ccc') }} />
            {language}
        </div>
    );
};

RepoLanguage.propTypes = {
    language: PropTypes.string,
};

RepoLanguage.defaultProps = {
    language: '',
};

export default RepoLanguage;
