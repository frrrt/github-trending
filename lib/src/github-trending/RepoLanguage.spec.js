import React from 'react';
import { render } from '@testing-library/react';
import RepoLanguage from './RepoLanguage';

const baseProps = {
    language: 'JavaScript',
};

describe('RepoLanguage', () => {
    it('is rendered correctly', () => {
        expect(render(<RepoLanguage {...baseProps} />).container).toMatchSnapshot();
    });

    it('no language given', () => {
        expect(render(<RepoLanguage />).container).toMatchSnapshot();
    });

    it('unknown language given', () => {
        expect(render(<RepoLanguage language="this-language-does-not-exist" />).container).toMatchSnapshot();
    });
});
