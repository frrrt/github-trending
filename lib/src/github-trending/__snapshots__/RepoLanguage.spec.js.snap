// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoLanguage is rendered correctly 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-circle-2"
      style="background-color: rgb(241, 224, 90);"
    />
    JavaScript
  </div>
</div>
`;

exports[`RepoLanguage no language given 1`] = `
<div>
  <div
    class="makeStyles-root-3"
  >
    <div
      class="makeStyles-circle-4"
    />
    
  </div>
</div>
`;

exports[`RepoLanguage unknown language given 1`] = `
<div>
  <div
    class="makeStyles-root-5"
  >
    <div
      class="makeStyles-circle-6"
      style="background-color: rgb(204, 204, 204);"
    />
    this-language-does-not-exist
  </div>
</div>
`;
